### ISSUE TITLE
Add `json` and `sql` template rendering to Grid View

### ISSUE THREAD

**Issue by nathadfield** [apache/airflow#39527]
### Apache Airflow version

2.9.1

### What happened?

In Airflow v2.9.1, the `Rendered Template` link is now missing from the `Details` tab when in `Grid` view.

### 2.8.4
![Screenshot 2024-05-09 at 16 18 01](https://github.com/apache/airflow/assets/967119/c9edfee6-f9b4-4df4-9a84-6f144201b471)

### 2.9.1
![Screenshot 2024-05-09 at 16 18 53](https://github.com/apache/airflow/assets/967119/30661365-e45d-4940-9452-1176a7029bf6)

### What you think should happen instead?

It is still possible to get to the rendered template via `More Details` but having this link here is much more convenient. 

### How to reproduce

Simply start an instance of Airflow 2.9.1.

### Operating System

n/a

### Versions of Apache Airflow Providers

n/a

### Deployment

Astronomer

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by tirkarthi** [comment]
It has been made as a separate section in 2.9.1 in this PR. I guess the removal was intentional.

https://github.com/apache/airflow/pull/39122

cc: @bbovenzi

**Comment by nathadfield** [comment]
@tirkarthi Hmm, that's a bit disappointing.  I hate clicking more than times that I have to, or was previously used to.

**Comment by tirkarthi** [comment]
Isn't rendered template available in details tab at the bottom? Is the new section missing some details from old page?

**Comment by nathadfield** [comment]
@tirkarthi Ah, ok, well, I suppose it depends on what is being rendered.  For example, SQL being run via the `BigQueryInsertJobOperator` looks like this when you go to the specific rendered template page which used to be on `Details`.

<img width="1578" alt="Screenshot 2024-05-10 at 14 40 28" src="https://github.com/apache/airflow/assets/967119/51cb21d4-8ee1-42d4-a374-e822ffdda9b9">

Whereas it looks like this now.

<img width="1042" alt="Screenshot 2024-05-10 at 14 40 46" src="https://github.com/apache/airflow/assets/967119/195d4671-a0f9-43f7-bce3-8cad99e9cb9f">

The important thing is that it's not possible to copy and paste the SQL directly from the UI.

**Comment by tirkarthi** [comment]
Thanks, the legacy page uses renderers along with pformat which indents the dict. Maybe the issue can updated to have formatting for the values like the legacy page.

**Comment by nathadfield** [comment]
@tirkarthi It's not just about dict, but also the SQL contained within it.  If the new page could render both in a similar way then great!

**Comment by bbovenzi** [comment]
Yes, we should render the json and sql better in the grid view. We should add a copy button too.

**Comment by nathadfield** [comment]
@bbovenzi Great thanks!  I'll change this to a feature request then.

**Comment by jscheffl** [comment]
I also always wanted to contribute this but never came to this. I'd favor making this consistent like the DAG Run Conf which is rendered as a struct with a copy option.

Same applies for XCom results, too.

**Comment by tirkarthi** [comment]
I was working on this issue and it looks complex since the legacy page uses the task template fields to get the relevant renderer and uses pygment lexer for highlighting along with lot of other inference for the objects. The issue is that in this case there is configuration which is a json and configuration.query.query which is a sql with both of them template fields. But in the API response the rendered template field string is just returned without any lexer information. Also the legacy page has access to dagbag and the corresponding task which the API endpoints don't have. 

I have made some progress on this issue and can raise a PR once I get to a working state.

Thanks

**Comment by tirkarthi** [comment]
Another thing is that case in the fields in the response like `bash_command` is turned to `bashCommand` at the react layer which gets rendered in the UI as `bashCommand` which is also incorrect and a bug IMO.

**Comment by tirkarthi** [comment]
I have created https://github.com/apache/airflow/pull/39918 as a possible solution. Thanks.

**Comment by bbovenzi** [comment]
Closing in favor of https://github.com/apache/airflow/issues/49064 as the issue to do this in 3.1
